stuffer/s2n_stuffer.c

```diff
@@ -23,10 +23,22 @@
 #include "utils/s2n_blob.h"
 #include "utils/s2n_mem.h"
 
+bool s2n_stuffer_is_valid(const struct s2n_stuffer* stuffer)
+{
+  return S2N_OBJECT_PTR_IS_READABLE(stuffer) && 
+    s2n_blob_is_valid(&stuffer->blob) &&
+    /* <= is valid because we can have a fully written/read stuffer */
+    stuffer->read_cursor <= stuffer->blob.size &&
+    stuffer->write_cursor <= stuffer->blob.size &&
+    stuffer->read_cursor <= stuffer->write_cursor;
+    /*any combination of wiped, alloced, growable, and tainted are valid */
+}
+
 int s2n_stuffer_init(struct s2n_stuffer *stuffer, struct s2n_blob *in)
 {
-    stuffer->blob.data = in->data;
-    stuffer->blob.size = in->size;
+    S2N_PRECONDITION(S2N_OBJECT_PTR_IS_WRITABLE(stuffer));
+    S2N_PRECONDITION(s2n_blob_is_valid(in));
+    stuffer->blob = *in;
```

Comment:
is this line a bug fix?

Reply:
Yes.  I should have called it out as such.  The old code set the `data` and `size` but not the `allocated` and `mlocked` fields